rclpy/rclpy/impl/rmw_implementation_tools.py

```diff
@@ -85,23 +83,8 @@ def import_rmw_implementation():
     if __rmw_implementation_module is not None:
         return __rmw_implementation_module
     available_implementations = get_rmw_implementations()
-    if __selected_rmw_implementation is None:
-        logger = logging.getLogger('rclpy')
-        # prefer FastRTPS, otherwise first in alphabetical order
-        # the same logic is implemented in
-        # rmw_implementation_cmake/cmake/get_default_rmw_implementation.cmake
-        default_impl = 'rmw_fastrtps_cpp'
-        if default_impl in available_implementations:
-            select_rmw_implementation(default_impl)
-        elif available_implementations:
-            select_rmw_implementation(available_implementations[0])  # select the first one
-        else:
-            raise NoImplementationAvailableException()
-        logger.debug("Implicitly selecting the '{0}' rmw implementation."
-                     .format(__selected_rmw_implementation))
-    module_name = '._rclpy__{rmw_implementation}'.format(
-        rmw_implementation=__selected_rmw_implementation,
-    )
+    __selected_rmw_implementation = 'rmw_implementation'
```

Comment:
> You will be able to select the rmw implementation the same way as for C++ by setting the environment variable `RMW_IMPLEMENTATION`.

That ignores the current use case which allows you to query the available ones, and then decide in Python code which one to use. I think that's actually pretty useful, that's why I suggested you look at whether or not the env variable mechanism can be used to preserve that functionality or that we just continue to build one for each. You didn't address this point at all and simply stated "We don't need to build separate Python modules anymore." which ignores the issue.

> There is certainly more cleanup possible since all this rmw impl. selection logic in Python can be removed. But I will very likely leave that to a follow up PR since the complexity of the current set of PRs is already pretty high.

-1. Sorry, I don't support merging this while creating dead code and leaving tests disabled.

In my opinion this pr needs to either justify why the existing functionality is no longer required and remove the dead code and tests related to the removed feature or fix the existing code so the feature continues to work with the new mechanism.